services/thermal_mgr/thermal_mgr.c

```diff
@@ -1,71 +1,101 @@
-  memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
-  thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
-
-  memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
-  memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
-
-  thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
-}
-
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
-}
-
-void osHandlerLM75BD(void) {
-  /* Implement this function */
-}
-
-static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
-  while (1) {
-    
-  }
-}
-
-void addTemperatureTelemetry(float tempC) {
-  printConsole("Temperature telemetry: %f deg C\n", tempC);
-}
-
-void overTemperatureDetected(void) {
-  printConsole("Over temperature detected!\n");
-}
-
-void safeOperatingConditions(void) { 
-  printConsole("Returned to safe operating conditions!\n");
-}
+#include "thermal_mgr.h"
+#include "errors.h"
+#include "lm75bd.h"
+#include "console.h"
+
+#include <FreeRTOS.h>
+#include <os_task.h>
+#include <os_queue.h>
+
+#include <string.h>
+//test
+#define THERMAL_MGR_STACK_SIZE 256U
+
+static TaskHandle_t thermalMgrTaskHandle;
+static StaticTask_t thermalMgrTaskBuffer;
+static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
+
+#define THERMAL_MGR_QUEUE_LENGTH 10U
+#define THERMAL_MGR_QUEUE_ITEM_SIZE sizeof(thermal_mgr_event_t)
+
+static QueueHandle_t thermalMgrQueueHandle;
+static StaticQueue_t thermalMgrQueueBuffer;
+static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
+
+static void thermalMgr(void *pvParameters);
+
+void initThermalSystemManager(lm75bd_config_t *config) {
+  memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
+  memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
+  
+  thermalMgrTaskHandle = xTaskCreateStatic(
+    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+
+  memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
+  memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
+
+  thermalMgrQueueHandle = xQueueCreateStatic(
+    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
+
+}
+
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+  /* Send an event to the thermal manager queue */
+  if(thermalMgrQueueHandle == NULL)
+    return ERR_CODE_INVALID_STATE;
+  else if(event == NULL)
+    return ERR_CODE_INVALID_ARG;
+  
+  if(xQueueSend(thermalMgrQueueHandle, (void*) event, 10) == pdPASS)
+    return ERR_CODE_SUCCESS;  
+
+  return ERR_CODE_SUCCESS;

```

Comment:
if we reach here then the xQueueSend was not successful due to the queue being full, so we should return an error code instead of success